### PR TITLE
Sema: Remove bogus call to requestNominalLayout()

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: test/multifile/require-layout-call-result.swift [hunk 1]
```diff
@@ -1,8 +1,8 @@
-// RUN: %target-typecheck-verify-swift -module-name test -primary-file %S/Inputs/require-layout-call-result-primary.swift
+// RUN: %target-swift-frontend -emit-ir -module-name test %s -primary-file %S/Inputs/require-layout-call-result-primary.swift
 
 
 class C<T> {
-  dynamic func broken() { } // expected-error{{'dynamic'}}
+  func broken() { }
```
[the remark this answers is not in the document]
Comment (DougGregor) [reply test/multifile/require-layout-call-result.swift]: Hmm. I don't have a test case that demonstrates the actual crash; this 

The logic behind my addition of this `requestNominalLayout` call (in 76eb767b970770f52eab5486a0434410d6cd216a) is that when you call a function, you produce a value of that type. IRGen is going to need to be able to lay out that type, so the type checker needs to be able to precompute it.

[the remark this answers is not in the document]
Comment (slavapestov) [reply test/multifile/require-layout-call-result.swift]: getAnyNominal() is not sufficient, though. You might return a tuple containing nominals, or optionals, etc, and extract fields from there without making further calls.

[the remark this answers is not in the document]
Comment (DougGregor) [reply test/multifile/require-layout-call-result.swift]: I agree that it's not sufficient.

**Pull request merged**